sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
 
 from azure.core.pipeline.policies import HttpLoggingPolicy
 
+from .http_constants import HttpHeaders
 
 
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs) -> None: # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
         if self._enable_diagnostics_logging:
```

Comment:
This current implementation only supports enabling diagnostics at construction, as there's currently not a clean way to override this per method. We could open an issue to look at adding support for that in future.

Reply:
not sure what you mean here - I have tried it and you can pass in the logger to just a method to get diagnostics for just that request

Reply:
Yes - you can definitely log only singular requests.
What's not currently supported is updating the header whitelist per request. So you can choose specific operations to log, but they will all log the same thing.
The header whitelist is currently an instance variable, and therefore it would be non-threadsafe to change it per request. We would need to update the Core policy to allow a per-request white list to be passed in or something similar.